### PR TITLE
[509] Fix macro expansion diagnostic for multi-binding variable declarations in MacroSystem

### LINKED ISSUE #2133: Diagnostic triggered for every multi-binding variable declaration
### Description

The `MacroApplicationError.accessorMacroOnVariableWithMultipleBindings` diagnostic error is leaking into all nested variable declarations that have multiple bindings.

For quick reference, here's the invalid usage that merits the error:

```
@constantOneGetter
var x: Int, y: Int
┬─────────────────
╰─ 🛑 swift-syntax applies macros syntactically and there is no way to represent a variable
      declaration with multiple bindings that have accessors syntactically. While the compiler 
      allows this expansion, swift-syntax cannot represent it and thus disallows it.
```

Here are two cases where the error is erroneously triggered:

```
@Test
struct S {
  var x: Int, y: Int
  ┬─────────────────
  ╰─ 🛑 swift-syntax … no way to represent a variable declaration with multiple bindings…
}

@Test
struct Q {
  struct R {
    var i: Int, j: Int
    ┬─────────────────
    ╰─ 🛑 swift-syntax … no way to represent a variable declaration with multiple bindings…
  }
}
```

### Steps to Reproduce

Add these two two failing tests to `MemberMacroTests` (failing on `main`, I tested 71b7378): 

```swift
// ...

  func testStructVariableDeclWithMultipleBindings() {
    struct TestMacro: MemberMacro {
      static func expansion(
        of node: AttributeSyntax,
        providingMembersOf declaration: some DeclGroupSyntax,
        in context: some MacroExpansionContext
      ) throws -> [DeclSyntax] {
        return []
      }
    }

    assertMacroExpansion(
      """
      @Test
      struct S {
        var x: Int, y: Int
      }
      """,
      expandedSource: """
      struct S {
        var x: Int, y: Int
      }
      """,
      macros: ["Test": TestMacro.self],
      indentationWidth: indentationWidth
    )
  }

  func testNestedStructVariableDeclWithMultipleBindings() {
    struct TestMacro: MemberMacro {
      static func expansion(
        of node: AttributeSyntax,
        providingMembersOf declaration: some DeclGroupSyntax,
        in context: some MacroExpansionContext
      ) throws -> [DeclSyntax] {
        return []
      }
    }

    assertMacroExpansion(
      """
      @Test
      struct Q {
        struct R {
          var i: Int, j: Int
        }
      }
      """,
      expandedSource: """
      struct Q {
        struct R {
          var i: Int, j: Int
        }
      }
      """,
      macros: ["Test": TestMacro.self],
      indentationWidth: indentationWidth
    )
  }

// ...
```

### PR DESCRIPTION
**Pull request by gohanlon**
Cherry-pick #2139 to package-release/509.

---

When there are no macro attributes attached to the visited `VariableDeclSyntax` node, proceed without performing any additional checks or transformations.

- Addressed the erroneous triggering of `MacroApplicationError.accessorMacroOnVariableWithMultipleBindings` diagnostic for nested multi-binding variable declarations.
- Added corresponding tests to validate the fix.
- Introduce fileprivate `NoOpMemberMacro`, extracted from `testCommentAroundeAttachedMacro` and used in 2 new tests.

Fixes #2133
rdar://114836887

### CODE REVIEW

**Comment by ahoppen** [comment]
Thanks. I want to get a second review on this from @bnbarham and then we can merge it.

**Comment by gohanlon** [comment]
I reformatted with `format.py` and amended both this PR (and #2139).

**Review by bnbarham — APPROVED** [review]

**Pull request merged**
